impl/src/main/java/com/okta/sdk/impl/client/DefaultClientBuilder.java

```diff
@@ -353,10 +356,15 @@ private void validateOAuth2ClientConfig(ClientConfiguration clientConfiguration)
         Assert.notNull(clientConfiguration.getClientId(), "clientId cannot be null");
         Assert.isTrue(clientConfiguration.getScopes() != null && !clientConfiguration.getScopes().isEmpty(),
             "At least one scope is required");
-        Assert.notNull(clientConfiguration.getPrivateKey(), "privateKey cannot be null");
-        Path privateKeyPemFilePath = Paths.get(clientConfiguration.getPrivateKey());
-        boolean privateKeyPemFileExists = Files.exists(privateKeyPemFilePath, new LinkOption[]{ LinkOption.NOFOLLOW_LINKS });
-        Assert.isTrue(privateKeyPemFileExists, "privateKey file does not exist");
+        String privateKey = clientConfiguration.getPrivateKey();
+        Assert.hasText(privateKey, "privateKey cannot be null (either PEM file path (or) full PEM content must be supplied)");
+
+        if (!(privateKey.startsWith("-----BEGIN PRIVATE KEY-----") && privateKey.contains("-----END PRIVATE KEY-----"))) {
```

Comment:
There are a few different types of PEM wrapper strings (IIRC between PKCS#1 and PKCS#8), checking just the start string startsWith `-----BEGIN` (should be enough to cover most of the edge cases, though if you want a more concrete solution, take a look at what bouncy castle does

Reply:
Sure, bouncy castle does exactly what you mentioned (check startsWith `-----BEGIN`). Updated my code to do the same per your recommendation.